tutorial/01_images_are_arrays.md

```diff
@@ -0,0 +1,356 @@
+linear0 = np.linspace(0, 1, 2500).reshape((50, 50))
+linear1 = np.linspace(0, 255, 2500).reshape((50, 50)).astype(np.uint8)
+
+print("Linear0:", linear0.dtype, linear0.min(), linear0.max())
+print("Linear1:", linear1.dtype, linear1.min(), linear1.max())
+
+fig, (ax0, ax1) = plt.subplots(1, 2, figsize=(15, 15))
+ax0.imshow(linear0, cmap='gray')
+ax1.imshow(linear1, cmap='gray');
+```
+
+When we first designed the library, we always *assumed* that floating type images go from 0-1, and unsigned bytes from 0 to 255.
+
+For floating point images, we're moving away from that design, because often you want to represent quantities that don't fit that mold: e.g., you may be looking at temperature or rainfall data.
+
+If you're just working with standard imaging data, continue to use 0-1.
+
+When loading integer images (e.g., 0-255), you'll often want to convert those to floating point (0-1). You may do that using `img_as_float`:
+
+```python
+cat = ski.data.chelsea()
+print(cat.dtype, cat.min(), cat.max())
+
+cat_float = ski.util.img_as_float(cat)
+print(cat_float.dtype, cat_float.min(), cat_float.max())
+
+print()
+print("231/255 =", 231/255.)
+```
+
+## Image I/O
+
+Mostly, we won't be using images from the scikit-image example data sets, but images stored on disk in JPEG, PNG, or TIFF format.  Since scikit-image operates on NumPy arrays, *any* image reader library that returns arrays will do.  We recommend `imageio`, but `matplotlib`, `pillow`, etc. also work.
+
+scikit-image provides a convenience wrapper around `image`, in the form of the `skimage.io` submodule:
+
+```python
+image = ski.io.imread('./data/balloon.jpg')
+
+print(type(image))
+print(image.dtype)
+print(image.shape)
+print(image.min(), image.max())
+
+plt.imshow(image);
+```
+
+We also have the ability to load multiple images from a folder:
+
+```python
+ic = ski.io.ImageCollection(['./*.png', './data/*.png', './data/*.jpg'])
+
+print('Type:', type(ic))
+
+ic.files
+```
+
+```python
+import os
+
+def plot_collection(ic : ski.io.ImageCollection):
+    """Display images in an ImageCollection in a grid.
+
+    """
+    f, axes = plt.subplots(ncols=3, nrows=int(np.ceil(len(ic) / 3)))
+
+    # subplots returns the figure and an array of axes
+    # we use `axes.ravel()` to turn these into a list
+    axes = axes.ravel()
+
+    for ax in axes:
+        # Hide axis labels
+        ax.axis('off')
+
+    for i, image in enumerate(ic):
+        axes[i].imshow(image, cmap='gray')
+        axes[i].set_title(os.path.basename(ic.files[i]))
+    
+    plt.tight_layout()
+```
+
+```python
+plot_collection(ic)
+```
+
+<!-- #region jp-MarkdownHeadingCollapsed=true -->
+### Aside: `enumerate`
+
+`enumerate` gives us each element in a container, along with its position.
+<!-- #endregion -->
+
+```python
+animals = ['cat', 'dog', 'leopard']
+```
+
+```python
+for i, animal in enumerate(animals):
+    print('The animal in position {} is {}'.format(i, animal))
+```
+
+## Exercise 1: draw the letter H
```

Comment:
I would suggest removing this exercise completely to trim the notebook. While it shows some basic NumPy slicing, I feel like the other exercises are more useful to retain.

Reply:
Oh, yes! That was my intention.